### PR TITLE
win32 adjustments

### PR DESCRIPTION
**Pull request by ace20022**
@afedchin @FernetMenta  please have a look, I'm not very familiar with that stuff. I'm especially unsure about the interface additions.

Currently only files without an audio track work. Otherwise the audio queue is stuck at 0%, but I had no time to investigate why.

### CODE REVIEW

File: xbmc/rendering/dx/RenderSystemDX.cpp [hunk 1]
```diff
@@ -425,8 +425,8 @@ void CRenderSystemDX::DeleteDevice()
     m_pGUIShader->End();
 
   // tell any shared resources
-  for (std::vector<ID3DResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
-    (*i)->OnDestroyDevice();
+  for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
+    (*i)->OnDestroyDisplay();
```
Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: this wrong. All ID3DResource still await OnDestroyDevice to destroy any device related resources like a textures, shaders and so one.

File: xbmc/rendering/dx/RenderSystemDX.cpp [hunk 2]
```diff
@@ -479,8 +479,8 @@ void CRenderSystemDX::OnDeviceLost()
   else
   {
     // just resetting the device
-    for (std::vector<ID3DResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
-      (*i)->OnLostDevice();
+    for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
+      (*i)->OnLostDisplay();
```
Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: same as above

File: xbmc/rendering/dx/RenderSystemDX.cpp [hunk 3]
```diff
@@ -492,8 +492,8 @@ void CRenderSystemDX::OnDeviceReset()
     CreateDevice();
   else
   { // we're back
-    for (std::vector<ID3DResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
-      (*i)->OnResetDevice();
+    for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
+      (*i)->OnResetDisplay();
```
Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: same as above

File: xbmc/rendering/dx/RenderSystemDX.cpp [hunk 4]
```diff
@@ -694,8 +694,8 @@ bool CRenderSystemDX::CreateDevice()
   m_needNewDevice = false;
 
   // tell any shared objects about our resurrection
-  for (std::vector<ID3DResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
-    (*i)->OnCreateDevice();
+  for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
+    (*i)->OnCreateDisplay();
```
Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: same as above

**Comment by afedchin** [comment]
this will work until any resetting of device like a transition from one adapter to another.

**Comment by afedchin** [comment]
here is my vision of this https://github.com/afedchin/xbmc/commit/6fa0312c1466c4ab4201c3cc29db2e5fbef83f0d

**Comment by FernetMenta** [comment]
IDispResource is no substitute for D3DResource. IDispResource is Kodi related while D3DResource is pure Windows.
@afedchin 's version looks ok after a brief glance

**Comment by ace20022** [comment]
thanks

**Pull request closed without merging**